docs/backends/app/backend_info_app.py

```diff
@@ -21,10 +22,67 @@
 ibis.options.verbose_log = lambda sql: sql_queries.append(sql)
 
 
+def get_single_query_param(key: str, default=None):
+    query_param_list = st.experimental_get_query_params().get(key)
+    if query_param_list:
+        return query_param_list[0]
+    return default
+
+
+def github_paginated_request(url):
+    next_url = url
+    resources = []
+    headers = {
+        'X-GitHub-Api-Version': '2022-11-28',
+        'Accept': 'application/vnd.github+json',
+    }
+    # On Streamlit Cloud, token should be stored as secret.
+    # See:
+    # https://docs.streamlit.io/streamlit-community-cloud/get-started/deploy-an-app/connect-to-data-sources/secrets-management
+    github_token = os.environ.get('GITHUB_TOKEN')
```

Comment:
What permissions does this token need?

Reply:
I don't think any are needed because we're connecting to public repositories only.